lib/matplotlib/backends/_backend_tk.py

```diff
@@ -652,6 +652,24 @@ def _set_image_for_button(self, button):
         # Use the high-resolution (48x48 px) icon if it exists and is needed
         with Image.open(path_large if (size > 24 and path_large.exists())
                         else path_regular) as im:
+            colors = ["background", "activebackground"]
+            if isinstance(button, tk.Checkbutton):
+                colors.append("selectcolor")
+            for color in colors:
+                if button.cget(color) == "":
+                    continue
+                # Get the color as an (r, g, b) tuple in the range from 0 to
+                # 65535, and check if its HSV V is under 50%.
+                if max(button.winfo_rgb(button.cget(color))) < 65535 / 2:
+                    # Change the colour of the icon to be visible against
+                    # the dark background.
+                    foreground = (255 / 65535) * np.array(
+                        button.winfo_rgb(button.cget("foreground")))
+                    image_data = np.asarray(im)
```

Comment:
```suggestion
                    image_data = np.asarray(im).copy()
```
Otherwise, I get the following error with Pillow 8.1.2:
```
import matplotlib
matplotlib.use('tkagg')
import matplotlib.pyplot as plt
plt.plot([1, 3, 2])
plt.show()
```

```
Traceback (most recent call last):
  File "te.py", line 4, in <module>
    plt.plot([1, 3, 2])
  File "/home/tim/dev/matplotlib/lib/matplotlib/pyplot.py", line 2764, in plot
    return gca().plot(
  File "/home/tim/dev/matplotlib/lib/matplotlib/pyplot.py", line 2269, in gca
    return gcf().gca(**kwargs)
  File "/home/tim/dev/matplotlib/lib/matplotlib/pyplot.py", line 851, in gcf
    return figure()
  File "/home/tim/dev/matplotlib/lib/matplotlib/pyplot.py", line 795, in figure
    FigureClass=FigureClass, **kwargs)
  File "/home/tim/dev/matplotlib/lib/matplotlib/pyplot.py", line 311, in new_figure_manager
    return _backend_mod.new_figure_manager(*args, **kwargs)
  File "/home/tim/dev/matplotlib/lib/matplotlib/backend_bases.py", line 3438, in new_figure_manager
    return cls.new_figure_manager_given_figure(num, fig)
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 954, in new_figure_manager_given_figure
    manager = cls.FigureManager(canvas, num, window)
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 413, in __init__
    self.toolbar = self._get_toolbar()
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 435, in _get_toolbar
    toolbar = NavigationToolbar2Tk(self.canvas, self.window)
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 540, in __init__
    command=getattr(self, callback),
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 694, in _Button
    NavigationToolbar2Tk._set_image_for_button(self, b)
  File "/home/tim/dev/matplotlib/lib/matplotlib/backends/_backend_tk.py", line 670, in _set_image_for_button
    image_data[black_mask, :3] = foreground
ValueError: assignment destination is **read-only**
```

Reply:
Thanks, didn't realise it worked different in older Pillow versions!